### ISSUE TITLE
Netdata updater folders on / directory

### ISSUE THREAD

**Issue by samuelmf** [netdata/netdata#8128]
Hi i have a Centos 7 server and installed netdata but i've noted folders in my / directory 
https://i.imgur.com/S77a82T.png
Take a look at the link

Is that normal?

**Comment by prologic** [comment]
No that is not normal. Looking...

**Comment by prologic** [comment]
I found where we're doing this... This has been around since 1st Mar 2019 from PR #5487 and this behaviour seems to happen when we cannot find `/tmp` mounted in some "noexec" form or something (_not quite 100% sure on the logic there at this minute..._)

**Comment by prologic** [comment]
The other related PR being #5232 I don't fully understand what happened here. It seems like we originally wrote the logic incorrectly, fixed it then re-broke it? wut?!

**Comment by prologic** [comment]
@samuelmf Can you confirm a few things for me?

- Your previous and current version of NetData
- A copy of your cron job -- I assume this is running from cron
- The output of `mount` and/or `cat /proc/mounts`

Meanwhile are you okay to manually / scripted / automated clean these directories up? Really sorry about the inconvenience caused here; this looks like a long standing bug/issue. I will dig further into this today...

**Comment by samuelmf** [comment]
- Your previous and current version of NetData How is possible to determine that, because i have the auto update enabled.

For cronjob i only see a job programmed on the daily folder.
netdata-updater -> /usr/libexec/netdata/netdata-updater.sh


**cat /proc/mounts results**
```
[root@core cron.daily]# cat /proc/mounts
rootfs / rootfs rw 0 0
sysfs /sys sysfs rw,seclabel,nosuid,nodev,noexec,relatime 0 0
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
devtmpfs /dev devtmpfs rw,seclabel,nosuid,size=3990116k,nr_inodes=997529,mode=755 0 0
securityfs /sys/kernel/security securityfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /dev/shm tmpfs rw,seclabel,nosuid,nodev 0 0
devpts /dev/pts devpts rw,seclabel,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 0 0
tmpfs /run tmpfs rw,seclabel,nosuid,nodev,mode=755 0 0
tmpfs /sys/fs/cgroup tmpfs ro,seclabel,nosuid,nodev,noexec,mode=755 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd 0 0
pstore /sys/fs/pstore pstore rw,nosuid,nodev,noexec,relatime 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpuacct,cpu 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,nosuid,nodev,noexec,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset 0 0
cgroup /sys/fs/cgroup/hugetlb cgroup rw,nosuid,nodev,noexec,relatime,hugetlb 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup rw,nosuid,nodev,noexec,relatime,net_prio,net_cls 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,nosuid,nodev,noexec,relatime,pids 0 0
configfs /sys/kernel/config configfs rw,relatime 0 0
/dev/mapper/sys--t8w7-root / ext4 rw,seclabel,relatime,data=ordered,jqfmt=vfsv0,usrjquota=quota.user 0 0
selinuxfs /sys/fs/selinux selinuxfs rw,relatime 0 0
debugfs /sys/kernel/debug debugfs rw,relatime 0 0
mqueue /dev/mqueue mqueue rw,seclabel,relatime 0 0
hugetlbfs /dev/hugepages hugetlbfs rw,seclabel,relatime 0 0
/dev/sda1 /boot ext4 rw,seclabel,relatime,data=ordered 0 0
/dev/mapper/sys--t8w7-tmp /tmp ext4 rw,seclabel,nosuid,noexec,relatime,data=ordered 0 0
/dev/mapper/sys--t8w7-tmp /var/tmp ext4 rw,seclabel,nosuid,noexec,relatime,data=ordered 0 0
tmpfs /run/user/1000 tmpfs rw,seclabel,nosuid,nodev,relatime,size=800172k,mode=700,uid=1000,gid=1000 0 0
tmpfs /run/user/995 tmpfs rw,seclabel,nosuid,nodev,relatime,size=800172k,mode=700,uid=995,gid=993 0 0
tmpfs /run/user/0 tmpfs rw,seclabel,nosuid,nodev,relatime,size=800172k,mode=700 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs rw,relatime,fd=30,pgrp=1,timeout=0,minproto=5,maxproto=5,direct 0 0
tmpfs /run/user/991 tmpfs rw,seclabel,nosuid,nodev,relatime,size=800172k,mode=700,uid=991,gid=988 0 0
[root@core cron.daily]# 

```

Sorry for the daly i was out of my city
Thanks for reply

**Comment by prologic** [comment]
@samuelmf Apologies for the delay on this one, other things got in the way.

I have a few questions for you now that I'm back on this and plan to complete it today.

- Why do you have your setup setup in such a way that `/tmp` is an ext4 partition backed by a disk and with `noexec` mount options? (_I'm mostly curious_)

- Can you (_if the logs are still available_) have a poke around your cron logs for any signs of the netdata updater script failing early or otherwise bailing out that would cause the following line not to execute?

https://github.com/netdata/netdata/blob/2c8cbc1f6336814cafdc8ea0043bbf936497e8ff/packaging/installer/netdata-updater.sh#L190

**Comment by prologic** [comment]
I can also reproduce this artificially by re-mounting `/tmp` with `noexec` and commenting out this line.

**Comment by Ferroin** [comment]
> * Why do you have your setup setup in such a way that `/tmp` is an ext4 partition backed by a disk and with `noexec` mount options? (_I'm mostly curious_)

I can't substantiate any specific reason to have `/tmp` backed by a physical partition, but mounting it with `noexec` is a rather common security choice. Most legitimate software never needs to execute files in `/tmp`, and just outright blocking that actually breaks quite a few older rootkits and worms.

**Comment by samuelmf** [comment]
Sorry for the delay guys, this situation of the covid is complicating everything here.
Regarding to question:

> Why do you have your setup setup in such a way that /tmp is an ext4 partition backed by a disk and with noexec mount options? (I'm mostly curious)

Really i dont know, the provisioning team of my server provider do that, i dont have a lot of experience in linux to know that, is taht fixable.

Where i can view for netdata update logs?

**Comment by prologic** [comment]
@samuelmf Its okay. I have a PR open to fix this anyway. Once its merged and you updated to a new nightly you should see this problem go away.

**Comment by samuelmf** [comment]
I will try to update to the last version and check if the problem dissapears

**Comment by prologic** [comment]
The PR isn't merged yet; I'll get it merged shortly! Sorry for the ddelay!

**Comment by prologic** [comment]
The fix in #8414 was merge ~7 days ago but I seemingly forgot to add the "Fixes #..." text in the PR :/ Sorry about that!
